projects/swimlane/ngx-graph/src/lib/graph/graph.component.ts

```diff
@@ -1309,6 +1318,36 @@ export class GraphComponent implements OnInit, OnChanges, OnDestroy, AfterViewIn
     return null;
   }
 
+  /**
+   * Checks if the graph has dimensions
+   */
+  public hasGraphDims(): boolean {
+    return this.graphDims.width > 0 && this.graphDims.height > 0;
+  }
+
+  /**
+   * Checks if all nodes have dimension
+   */
+  public hasNodeDims(): boolean {
+    return this.graph.nodes?.filter(node => node.dimension.width === 0 && node.dimension.height === 0).length === 0;
```

Comment:
nit pick: [Array.every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) could be used here

Reply:
Fixed